packages/world/script/migrations/1750380758-ƒix-seed-tracking.s.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import {console} from "forge-std/console.sol";
+
+import {IndexerResult, indexer} from "../utils/Indexer.sol";
+import {Config, config} from "../utils/config.sol";
+
+import {ResourceCount} from "../../src/codegen/tables/ResourceCount.sol";
+import {ObjectType, ObjectTypes} from "../../src/types/ObjectType.sol";
+
+import {DustScript} from "../DustScript.sol";
+
+contract FixSeedTracking is DustScript {
+    mapping(ObjectType => uint256) private totalCounts;
+
+    function run() public {
+        Config memory cfg = config();
+
+        // Define seed and sapling types using ObjectTypes library
+        ObjectType[11] memory seedsAndSaplings = [
+            // Seeds
+            ObjectTypes.WheatSeed,
+            ObjectTypes.PumpkinSeed,
+            ObjectTypes.MelonSeed,
+            // Saplings
+            ObjectTypes.OakSapling,
+            ObjectTypes.BirchSapling,
+            ObjectTypes.JungleSapling,
+            ObjectTypes.SakuraSapling,
+            ObjectTypes.AcaciaSapling,
+            ObjectTypes.SpruceSapling,
+            ObjectTypes.DarkOakSapling,
+            ObjectTypes.MangroveSapling
+        ];
+
+        // Build string for SQL queries
+        string memory seedTypes = "";
+        for (uint256 i = 0; i < seedsAndSaplings.length; i++) {
+            seedTypes = string.concat(seedTypes, vm.toString(seedsAndSaplings[i].unwrap()));
+            if (i < seedsAndSaplings.length - 1) {
+                seedTypes = string.concat(seedTypes, ", ");
+            }
+        }
+
+        // Query 1: Get planted seeds/saplings from EntityObjectType
+        console.log("\nCurrent Planted tracking:");
+        IndexerResult memory plantedResult = indexer(cfg).query(
+            string.concat(
+                "SELECT objectType, COUNT(*) as count FROM EntityObjectType WHERE objectType IN (",
```

Comment:
hmm does this mean it's necessarily planted?

Reply:
Even if its in your inventory it has the `EntityObjectType`

Reply:
If it is in your inventory it is not an entity

Reply:
ahh cuz if you mine it, the entity becomes air